neuralprophet/time_net.py

```diff
@@ -187,16 +187,14 @@ def __init__(
         # Covariates
         self.config_covar = config_covar
         if self.config_covar is not None:
-            assert self.n_lags > 0
             self.covar_nets = nn.ModuleDict({})
             for covar in self.config_covar.keys():
                 covar_net = nn.ModuleList()
-                d_inputs = self.n_lags
-                if self.config_covar[covar].as_scalar:
-                    d_inputs = 1
+                d_inputs = self.config_covar[covar].n_covars
                 for i in range(self.num_hidden_layers):
-                    covar_net.append(nn.Linear(d_inputs, self.d_hidden, bias=True))
-                    d_inputs = self.d_hidden
+                    d_hidden = d_inputs + n_forecasts if d_hidden is None else d_hidden
```

Comment:
let's make this (separate from AR):
if d_hidden is None:
    self.d_hidden = max(4, (n_lags + n_forecasts) /(2.0* num_hidden_layers)
else:
    self.d_hidden = d_hidden

Reply:
Same comment as above:
                    d_hidden = max(4, round((self.config_covar[covar].n_lags + n_forecasts)/(2.0*(num_hidden_layers+1)))) if d_hidden is None else d_hidden
replaced n_lags with self.config_covar[covar].n_lags so it does not depend on the AutoRegression lags